src/v/storage/tests/log_retention_tests.cc

```diff
@@ -124,6 +124,10 @@ FIXTURE_TEST(retention_test_size_time, gc_fixture) {
         offset += model::offset(num_records);
     }
 
+    // there is only an active segment. nothing should be reclaimable
```

Comment:
You asked me about this out of band, but we don't preserve the invariant that one segment is always on disk.

Reply:
cool, thanks. didn't know that.